plugin/src/main/java/io/jenkins/plugins/analysis/warnings/YoctoScanner.java

```diff
@@ -0,0 +1,46 @@
+package io.jenkins.plugins.analysis.warnings;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.jenkinsci.Symbol;
+import hudson.Extension;
+
+import io.jenkins.plugins.analysis.core.model.AnalysisModelParser;
+
+/**
+ * Provides a parser and customized messages for Yocto Scanner CLI (scannercli) reports.
+ *
+ * @author Michael Trimarchi
+ */
+public class YoctoScanner extends AnalysisModelParser {
+    private static final long serialVersionUID = 1L;
+    private static final String ID = "yoctocli";
+
+    /**
+     * Creates a new instance of {@link YoctoScanner}.
+     */
+    @DataBoundConstructor
+    public YoctoScanner() {
+        super();
+        // empty constructor required for stapler
+    }
+
+    /** Descriptor for this static analysis tool. */
+    @Symbol("yoctoScanner")
+    @Extension
+    public static class Descriptor extends AnalysisModelParserDescriptor {
+        /** Creates the descriptor instance. */
+        public Descriptor() {
+            super(ID);
+        }
+
+        @Override
+        public boolean canScanConsoleLog() {
+            return false;
+        }
+
+        @Override
+        public boolean isPostProcessingEnabled() {
+            return false;
+        }
+    }
+}
```

Comment:
What about adding another icon? The default warning icon might be inappropriate for security warnings? 

We can select from: https://fontawesome.com/icons/categories/security or https://ionic.io/ionicons

Example code to change the icon: https://github.com/jenkinsci/warnings-ng-plugin/blob/main/plugin/src/main/java/io/jenkins/plugins/analysis/warnings/ErrorProne.java

Reply:
```
diff --git a/plugin/pom.xml b/plugin/pom.xml
index 87a6b108c..b61c37f9f 100644
--- a/plugin/pom.xml
+++ b/plugin/pom.xml
@@ -25,7 +25,7 @@
     <changelist>-SNAPSHOT</changelist>
     <module.name>${project.groupId}.warnings.ng</module.name>
 
-    <analysis-model-api.version>12.4.0</analysis-model-api.version>
+    <analysis-model-api.version>12.5.0</analysis-model-api.version>
     <analysis-model-tests.version>${analysis-model-api.version}</analysis-model-tests.version>
     <pull-request-monitoring.version>335.v525cd64ec76b_</pull-request-monitoring.version>
 ```
 This for the pom
 ```
diff --git a/plugin/src/main/java/io/jenkins/plugins/analysis/warnings/YoctoScanner.java b/plugin/src/main/java/io/jenkins/plugins/analysis/warnings/YoctoScanner.java
index c41df8dc4..5e121760f 100644
--- a/plugin/src/main/java/io/jenkins/plugins/analysis/warnings/YoctoScanner.java
+++ b/plugin/src/main/java/io/jenkins/plugins/analysis/warnings/YoctoScanner.java
@@ -5,6 +5,7 @@ import org.jenkinsci.Symbol;
 import hudson.Extension;
 
 import io.jenkins.plugins.analysis.core.model.AnalysisModelParser;
+import io.jenkins.plugins.analysis.core.model.SymbolIconLabelProvider;
 
 /**
  * Provides a parser and customized messages for Yocto Scanner CLI (scannercli) reports.
@@ -42,5 +43,10 @@ public class YoctoScanner extends AnalysisModelParser {
         public boolean isPostProcessingEnabled() {
             return false;
         }
+
+        @Override
+        public StaticAnalysisLabelProvider getLabelProvider() {
+            return new SymbolIconLabelProvider(getId(), getDisplayName(), getDescriptionProvider(), "symbol-solid/bug plugin-font-awesome-api");
+        }
```
how icon are loaded?

Reply:
Icons are part of a different plug-in and are referenced in the UI. 

The bug is used for bug finders so you better should use a different one.

Reply:
https://fontawesome.com/icons/shield-exclamation?f=sharp&s=solid

Reply:
You need to choose a free one, there is a checkbox

Reply:
https://fontawesome.com/icons/shield-halved?f=classic&s=solid

Reply:
@uhafner Add the icon and tested, include the pom.xml change
![Screenshot from 2024-09-15 21-32-28](https://github.com/user-attachments/assets/60187bc4-4f67-47b0-acfa-9e737b7aa119)